MetaMask/metamask-mobile
# [FIX] Crash when reject two times connect wallet on in app browser

**Description**
The problem was when we press twice outside of the connection modal on our in-app browser, we were rejecting two times the same pending connection.

**Techincal reason**
On the controllers, we reset the id of the pending connection but on the state of that app, we didn't reset it. Because of that, we were trying to send a connection id that no longer existed on our pending connection list, and the app crashed.

**Proposed Solution**
Whenever we reject a pending connection request, we set the state that holds the information of that connection request to null (it will be filled when we try to do another wallet connect)
 
**Screenshots/Recordings**
https://recordit.co/cIyLwsZrvL

**Test Cases**
* Go to setting
* Press Security and Privacy
* Clear browser cookies 
* Clear privacy data
* Go to the in-app browser and browse for [opensea.io](https://opensea.io/)
* Open the modal of connecting your wallet
* Reject the connection by tapping twice outside of the connection modal
* Open again the connection modal and reject the connection
* Open again the connection modal and connect
* Everything should work as expected
* Tested with other dapps (sushi swap and pancake swap)


_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #5106 

**Checklist**
